app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryModel.kt

```diff
@@ -71,4 +78,12 @@ data class CustomDrugEntryModel(
   fun drugInfoProgressStateLoading(): CustomDrugEntryModel {
     return copy(drugInfoProgressState = IN_PROGRESS)
   }
+
+  fun buttonProgressStateIsSaving(): CustomDrugEntryModel {
+    return copy(saveButtonState = SAVING)
+  }
+
+  fun buttonProgressStateIsSaved(): CustomDrugEntryModel {
+    return copy(saveButtonState = SAVED)
+  }
```

Comment:
We can just have one function that says, `saveButtonStateChanged` and pass in the params right? 

```kt
fun saveButtonStateChanged(buttonState: ButtonState): CustomDrugEntryModel {
  return copy(saveButtonState = buttonState)
}
```

Reply:
umm yeah. We can do that in `drugInfoProgressStateLoading` as well right? Can you tell me how do we decide which one is the right approach for doing this in which scenario? I took this approach to maintain consistency in the model for progress states.

Reply:
I guess this is more of a personal preference than the correct approach. For example, if we decided to add another state to the button state, we would need to add another function here. Instead of that, we have been using a single function and just passing the state from the update function. 

You can refer few other places where we used `ButtonState`s.